drivers/spark-connector-new.md

```diff
@@ -0,0 +1,343 @@
+  .format("com.arangodb.spark")
+  .options(Map(
+    "password" -> "test",
+    "endpoints" -> "c1:8529,c2:8529,c3:8529",
+    "table" -> "users"
+  ))
+  .schema(new StructType(
+    Array(
+      StructField("likes", ArrayType(StringType, containsNull = false)),
+      StructField("birthday", DateType, nullable = true),
+      StructField("gender", StringType, nullable = false),
+      StructField("name", StructType(
+        Array(
+          StructField("first", StringType, nullable = true),
+          StructField("last", StringType, nullable = false)
+        )
+      ), nullable = true)
+    )
+  ))
+  .load()
+
+usersDF.filter(col("birthday") === "1982-12-15").show()
+```
+
+### Read Configuration
+
+- `database`: database name, default `_system`
+- `table`: datasource ArangoDB collection name, ignored if `query` is specified. Either `table` or `query` is required.
+- `query`: custom AQL read query. If set, `table` will be ignored. Either `table` or `query` is required.
+- `batchSize`: reading batch size, default `10000`
+- `sampleSize`: sample size prefetched for schema inference, only used if read schema is not provided, default `1000`
+- `fillBlockCache`: whether the query should store the data it reads in the RocksDB block cache (`true` or `false`), default `false`
+- `stream`: whether the query should be executed lazily, default `true`
+- `mode`: allows a mode for dealing with corrupt records during parsing:
+  - `PERMISSIVE` : when it meets a corrupted record, puts the malformed string into a field configured by 
+    `columnNameOfCorruptRecord`, and sets malformed fields to null. To keep corrupt records, a user can set a string 
+    type field named columnNameOfCorruptRecord in a user-defined schema. If a schema does not have the field, it drops 
+    corrupt records during parsing. When inferring a schema, it implicitly adds a `columnNameOfCorruptRecord` field in
+    an output schema
+  - `DROPMALFORMED`: ignores the whole corrupted records
+  - `FAILFAST`: throws an exception when it meets corrupted records
+- `columnNameOfCorruptRecord`: allows renaming the new field having malformed string created by `PERMISSIVE` mode
+
+### Predicate and Projection Pushdown
+
+The connector can convert some Spark SQL filters predicates into AQL predicates and push their execution down to the data source. In this way, ArangoDB can apply the filters and return only the matching documents.
+
+The following filter predicates (implementations of `org.apache.spark.sql.sources.Filter`) are pushed down:
+- `And`
+- `Or`
+- `Not`
+- `EqualTo`
+- `EqualNullSafe`
+- `IsNull`
+- `IsNotNull`
+- `GreaterThan`
+- `GreaterThanOrEqualFilter`
+- `LessThan`
+- `LessThanOrEqualFilter`
+- `StringStartsWithFilter`
+- `StringEndsWithFilter`
+- `StringContainsFilter`
+- `InFilter`
+
+Furthermore, the connector will push down also the subset of columns required by the Spark SQL query, so that only the relevant documents fields will be returned.
```

Comment:
```suggestion
Furthermore, the connector will push down also the subset of columns required by the Spark query, so that only the relevant documents fields will be returned.
```